Lib/test/test_dis.py

```diff
@@ -535,6 +579,17 @@ def test_disassemble_generator(self):
         gen_disas = self.get_disassembly(_g(1))  # Disassemble generator itself
         self.assertEqual(gen_disas, gen_func_disas)
 
+    def test_disassemble_async_generator(self):
+        agen_func_disas = self.get_disassembly(_ag) # Disassemble async generator function
+        agen_disas = self.get_disassembly(_ag(1)) # Disassemble async generator itself
+        self.assertEqual(agen_disas, agen_func_disas)
+
+    def test_disassemble_coroutine(self):
+        coro_func_disas = self.get_disassembly(_co) # Disassemble coroutine function
+        with ignore_coroutine_never_awaited_warnings():
+            coro_disas = self.get_disassembly(_co(1)) # Disassemble coroutine itself
```

Comment:
As per @1st1's comments above,  a simpler way of avoiding the warning here is to explicitly close the coroutine:

    coro = _co(1)
    coro.close()
    coro_disas = self.get_disassembly(coro)